### PR TITLE
[JN-828] Track viewed language in form responses

### PR DESCRIPTION
**Pull request by MatthewBemis**
#### DESCRIPTION

Updates the Answer table to store the code of the language that was being viewed when the participant submitted their responses. Also displays the language name in the form response admin view:

Screenshot 2024-02-13 at 12 55 30 PM

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*

Test via Populate:

* Reboot AdminApi + Admin UI
* Repopulate demo: `./scripts/populate_portal.sh demo`
* View the pre-reg, consent, and The Basics responses for Jonas Salk. Verify that you see a few "Answered in ..." displaying next to the questionStableId. Note that not all The Basics answers come pre-populated with language data: https://localhost:3000/demo/studies/heartdemo/env/sandbox/participants/HDSALK/consents/hd_hd_consent
* Verify no regressions in this view for a single-language study (i.e. OurHealth): https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/participants/OHSALK/surveys/oh_oh_basicInfo

Test with new participant:

* Reboot ParticipantApi + Participant UI (I suggest booting the UI with `REACT_APP_UNAUTHED_LOGIN=true HTTPS=true npm -w ui-participant start` so you can bypass B2C)
* Register as a new user in the Heart Demo study: https://sandbox.demo.localhost:3001
* Go to the Admin UI and view the form response data for your new participant
* Verify that you can see the language used when you filled out the form

### CODE REVIEW

File: core/src/main/resources/db/changelog/changesets/2024_02_09_answer_viewed_language.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+databaseChangeLog:
+  - changeSet:
+      id: add_answer_viewed_lang
+      author: mbemis
+      changes:
+        - addColumn:
+            tableName: answer
+            columns:
+              - column: { name: viewed_language, type: text, constraints: { nullable: true } }
```
Comment (devonbush) [inline core/src/main/resources/db/changelog/changesets/2024_02_09_answer_viewed_language.yaml]: I think these are easier to read if we don't specify nullable:true -- that's the default
```suggestion
              - column: { name: viewed_language, type: text }
```

**Review by devonbush — APPROVED** [review]
Looks good -- this really shows that you made the right move to put selectedLanguage in the useUser() context -- those parts of the code are really clean.  One super-minor nit

**Review by connorlbark — APPROVED** [review]
Works great and the code is super easy to follow! I did notice one thing, which is that when I answered in Spanish then changed the language and changed my answer, it still said I answered in Spanish. It's possible that is the intended behavior, though, so not blocking.

**Comment by MatthewBemis** [comment]
@connorlbark that's a great catch. fixed in https://github.com/broadinstitute/juniper/pull/737/commits/42620883897f31eed0cc61df9ef37e852d029a14

**Pull request merged**